sources/packages/web/src/components/common/OfferingForm.vue

```diff
@@ -83,13 +86,20 @@ export default defineComponent({
     },
     processing: {
       type: Boolean,
-      required: true,
+      required: false,
       default: false,
     },
   },
   setup(props, context) {
+    const data = ref({} as EducationProgramOfferingAPIOutDTO);
+    onMounted(async () => {
+      const offering =
+        await EducationProgramOfferingService.shared.getOfferingDetails(
+          props.offeringId,
```

Comment:
Since the `props.offeringId` can potentially be provided later is it not better to have the API call happening during the `whatchEffect`?